trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/security/SecurityFacade.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.s3.proxy.server.security;
+
+import java.util.Optional;
+
+public interface SecurityFacade
+{
+    String UPLOADS_ACTION = "uploads";
+    String DELETE_ACTION = "delete";
+
+    /**
+     *
+     * Actions are S3 operations such as CreateSession, PutBucketAcl, etc.
+     * see <a href="https://docs.aws.amazon.com/AmazonS3/latest/API/API_Operations.html">API Operations</a> for details.
+     * The "Request Syntax" for each API lists the action as the first query parameter
+     * E.g. <a href="https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteBucketIntelligentTieringConfiguration.html">DeleteBucketIntelligentTieringConfiguration</a>
+     * {@code /?intelligent-tiering&id=Id} - the action is {@code intelligent-tiering}.
+     */
+    SecurityResponse apply(String lowercaseHttpMethod, Optional<String> lowercaseAction, String region, String bucket, Optional<String> key);
```

Comment:
The identity is missing here. Should we pass in a `Credential` object representing the emulated credentials?

Reply:
The idea is that the actual implementation would contain whatever the caller needs. That's the purpose of `SecurityFacadeProvider`.

Reply:
Agreed, but then should we make it so that `SecurityFacadeProvider` takes in a `ParsedS3Request` and an `Identity`?

There is currently no way for the security implementation to know the identity of the requester (other than manually parsing the headers again)

Reply:
That would work, and probably is nice for complex use cases. But at least the external access key ID has already been parsed by in `SigningMetadata`, which is readily available at the time that we call this in `TrinoS3ProxyClient::proxyRequest`, do you think it'd be easy to pass it down from there?

Reply:
Sure, I can remove the values that are already in the ParsedRequest. That makes sense.

Reply:
Sorry - I missed the access Key ID. I'll add that.